test/corpus/declarations.txt

```diff
@@ -483,6 +494,55 @@ class Point {
               (decimal_integer_literal)
               (decimal_integer_literal))))))))
 
+=============================================
+Nested field expression in receiver parameter
+=============================================
+
+public class Inner4<T> {
```

Comment:
would be better Inner1, Inner2, Innter3, otherwise we have no idea why you start at 4.